plugin/loader/load_unix.go

```diff
@@ -5,52 +5,16 @@ package loader
 
 import (
 	"errors"
-	"fmt"
-	"os"
-	"path/filepath"
 	"plugin"
 
 	iplugin "github.com/ipfs/go-ipfs/plugin"
 )
 
 func init() {
-	loadPluginsFunc = linuxLoadFunc
+	loadPluginFunc = unixLoadPlugin
 }
 
-func linuxLoadFunc(pluginDir string) ([]iplugin.Plugin, error) {
-	var plugins []iplugin.Plugin
-
-	err := filepath.Walk(pluginDir, func(fi string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		if info.IsDir() {
-			if fi != pluginDir {
-				log.Warnf("found directory inside plugins directory: %s", fi)
-			}
-			return nil
-		}
-
-		if info.Mode().Perm()&0111 == 0 {
-			// file is not executable let's not load it
-			// this is to prevent loading plugins from for example non-executable
-			// mounts, some /tmp mounts are marked as such for security
-			log.Errorf("non-executable file in plugins directory: %s", fi)
-			return nil
-		}
-
-		if newPlugins, err := loadPlugin(fi); err == nil {
-			plugins = append(plugins, newPlugins...)
-		} else {
-			return fmt.Errorf("loading plugin %s: %s", fi, err)
-		}
-		return nil
-	})
-
-	return plugins, err
-}
-
-func loadPlugin(fi string) ([]iplugin.Plugin, error) {
+func unixLoadPlugin(fi string) ([]iplugin.Plugin, error) {
```

Comment:
Before, we wouldn't try loading plugins on platforms that didn't support plugins.

Now, we try loading plugins everywhere, but return an error if we find a plugin that we can't load (either because the platform is unsupported, or because something is wrong with the plugin).